.circleci/config.yml

```diff
@@ -10,6 +10,15 @@ references:
         user: builder
     working_directory: ~/epoch
 
+  merge_checkout: &merge_checkout
+    run:
+      name: Checkout merged master
+      command: |
+        if [ -n "$CIRCLE_PR_NUMBER" ]; then
+          git fetch origin +refs/pull/${CIRCLE_PR_NUMBER}/merge
+          git checkout -qf FETCH_HEAD
```

Comment:
Does this checkout step delete files from working copy if removed in master?

Reply:
Well, it should. It's a merge after all.

Also local changes are also removed because of the [`-f`](https://git-scm.com/docs/git-checkout#git-checkout--f) option.

Reply:
I didn't find this Github feature/hack documented anywhere, but I assume that it's just the merge commit of the PR into master.

```bash
dincho at iMac in ~/Projects/aeternity/epoch on PT-155429174_merged_master_checkout
$ git fetch origin +refs/pull/802/merge
remote: Counting objects: 1, done.
remote: Total 1 (delta 0), reused 0 (delta 0), pack-reused 0
Unpacking objects: 100% (1/1), done.
From github.com:aeternity/epoch
 * branch            refs/pull/802/merge -> FETCH_HEAD

dincho at iMac in ~/Projects/aeternity/epoch on PT-155429174_merged_master_checkout
$ git checkout -qf FETCH_HEAD

dincho at iMac in ~/Projects/aeternity/epoch on 4231483
$ git show 4231483
commit 423148336ed5db4c60580ec0219679426841d198 (HEAD)
Merge: f7aa14b 10ff7b4
Author: Dincho Todorov <dincho@users.noreply.github.com>
Date:   Wed Feb 28 10:45:01 2018 +0000

    Merge 10ff7b4938b68e752b31b8e006c3716f45060f90 into f7aa14b5124b1b88229574235a01a7abe4116ed0


```

Reply:
```
dincho at iMac in ~/Projects/aeternity/epoch on 4231483
$ git rev-parse origin/master
f7aa14b5124b1b88229574235a01a7abe4116ed0

dincho at iMac in ~/Projects/aeternity/epoch on 4231483
$ git rev-parse origin/PT-155429174_merged_master_checkout
10ff7b4938b68e752b31b8e006c3716f45060f90
```